library/ssl_ticket.c

```diff
@@ -522,7 +522,7 @@ int mbedtls_ssl_ticket_parse(void *p_ticket,
         }
 #endif
 
-        mbedtls_ms_time_t ticket_lifetime = ctx->ticket_lifetime * 1000;
+        mbedtls_ms_time_t ticket_lifetime = session->ticket_lifetime * 1000;
```

Comment:
What about the TLS 1.2 case?

Reply:
I will take a look. Maybe it can be fixed in another PR.

ticket_lifetime of session is only available in client side for TLS1.2 . I am not sure if it is reasonable now.

Reply:
Just take a quick look on RFC 5077, RFC 4346 and `ssl_ticket.c`. 

For the time being,  I think TLS 1.2  should do same improvement also( Not very sure, need more investigation. )

The lifetime is saved in `mbedtls_ssl_ticket_context` which is part of parse/write callback. That means the constraints might not be checked when the callbacks are not builtin functions. RFC 4346 and 5077 ask a 24 hours limitation( not very sure now). So I think we should do same improvement also.